app/src/main/java/com/codesoom/assignment/filters/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,45 @@
+package com.codesoom.assignment.filters;
+
+import com.codesoom.assignment.application.AuthenticationService;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+import security.UserAuthentication;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class JwtAuthenticationFilter extends BasicAuthenticationFilter {
+    private final AuthenticationService authenticationService;
+
+    public JwtAuthenticationFilter(AuthenticationManager authenticationManager,
+                                   AuthenticationService authenticationService) {
+        super(authenticationManager);
+        this.authenticationService = authenticationService;
+    }
+
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain chain)
+            throws IOException, ServletException {
+
+        String authorization = request.getHeader("Authorization");
```

Comment:
이런 곳에는 잊지 말고 final 을 붙여 줍시다.
```suggestion
        final String authorization = request.getHeader("Authorization");
```